core/src/main/java/fr/sncf/osrd/standalone_sim/SafetySpeed.kt

```diff
@@ -0,0 +1,99 @@
+package fr.sncf.osrd.standalone_sim
+
+import fr.sncf.osrd.api.FullInfra
+import fr.sncf.osrd.api.api_v2.standalone_sim.SimulationScheduleItem
+import fr.sncf.osrd.conflicts.TravelledPath
+import fr.sncf.osrd.railjson.schema.schedule.RJSTrainStop
+import fr.sncf.osrd.sim_infra.api.*
+import fr.sncf.osrd.sim_infra.impl.ChunkPath
+import fr.sncf.osrd.utils.DistanceRangeMap
+import fr.sncf.osrd.utils.distanceRangeMapOf
+import fr.sncf.osrd.utils.indexing.StaticIdx
+import fr.sncf.osrd.utils.indexing.StaticIdxList
+import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.Speed
+import fr.sncf.osrd.utils.units.kilometersPerHour
+import fr.sncf.osrd.utils.units.meters
+
+/**
+ * Compute safety speed ranges, areas where the train has a lower speed limit because of a scheduled
+ * stop. For details, see https://osrd.fr/en/docs/reference/design-docs/timetable/#modifiable-fields
+ * (or google "VISA (VItesse Sécuritaire d'Approche)" for resources in French)
+ */
+fun makeSafetySpeedRanges(
+    infra: FullInfra,
+    chunkPath: ChunkPath,
+    routes: StaticIdxList<Route>,
+    schedule: List<SimulationScheduleItem>
+): DistanceRangeMap<Speed> {
+    val rawInfra = infra.rawInfra
+    val zonePaths = routes.flatMap { rawInfra.getRoutePath(it) }
+    val zonePathStartOffset = getRoutePathStartOffset(rawInfra, chunkPath, zonePaths)
+    val signalOffsets = getSignalOffsets(infra, zonePaths, zonePathStartOffset)
+
+    val stopsWithSafetySpeed = schedule.filter { it.receptionSignal.isStopOnClosedSignal }
+
+    val res = distanceRangeMapOf<Speed>()
+    for (stop in stopsWithSafetySpeed) {
+        val nextSignalOffset =
```

Comment:
nit: how about adding a comment to explicit business logic choices?
```suggestion
        // Currently, safety speed is applied to the next signal no matter the sight distance
        val nextSignalOffset =
```